packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -38,63 +38,56 @@ library LibProving {
         uint256 blockId,
         TaikoData.BlockEvidence memory evidence
     ) internal {
+        if (
+            evidence.parentHash == 0 ||
+            evidence.blockHash == 0 ||
+            evidence.blockHash == evidence.parentHash ||
+            evidence.signalRoot == 0 ||
+            evidence.gasUsed == 0
+        ) revert L1_INVALID_EVIDENCE();
+
+        if (blockId <= state.lastVerifiedBlockId || blockId >= state.numBlocks)
+            revert L1_BLOCK_ID();
+
+        TaikoData.Block storage blk = state.blocks[
+            blockId % config.ringBufferSize
+        ];
+
+        // Check the metadata hash matches the proposed block's. This is
+        // necessary to handle chain reorgs.
+        if (blk.metaHash != evidence.metaHash)
+            revert L1_EVIDENCE_MISMATCH(blk.metaHash, evidence.metaHash);
+
         bool isOracleProof = evidence.prover == address(0);
 
         if (isOracleProof) {
             address oracleProver = resolver.resolve("oracle_prover", true);
             if (oracleProver == address(0)) revert L1_ORACLE_DISABLED();
-
             if (msg.sender != oracleProver) {
-                if (evidence.zkproof.data.length == 64) {
-                    uint8 v = uint8(evidence.zkproof.verifierId);
+                if (evidence.proof.length == 64) {
+                    uint8 v = uint8(evidence.verifierId);
```

Comment:
Why not put `v` in `proof` as well?

Reply:
no perticular reason, just want to reuse the verifiedId field. Otherwise, we'll have a zero there too.